lib/src/directus_websocket_subscription.dart

```diff
@@ -0,0 +1,103 @@
+import 'dart:convert';
+
+import 'package:directus_api_manager/directus_api_manager.dart';
+import 'package:directus_api_manager/src/model/directus_data.dart';
+import 'metadata_generator.dart';
+import 'package:reflectable/reflectable.dart';
+
+class DirectusWebSocketSubscription<Type extends DirectusData> {
+  final List<String>? fields;
+  final Filter? filter;
+  final List<SortProperty>? sort;
+  final String uid;
+  final int? limit;
+  final int? offset;
+
+  Function(Map<String, dynamic>)? onCreate;
+  Function(Map<String, dynamic>)? onUpdate;
+  Function(Map<String, dynamic>)? onDelete;
+
+  String get collection => collectionMetadata.endpointName;
+
+  final MetadataGenerator _metadataGenerator = MetadataGenerator();
+
+  ClassMirror get specificClass =>
+      _metadataGenerator.getClassMirrorForType(Type);
+  CollectionMetadata get collectionMetadata =>
+      _collectionMetadataFromClass(specificClass);
+
+  DirectusWebSocketSubscription(
+      {required this.uid,
+      this.fields,
+      this.filter,
+      this.sort,
+      this.limit,
+      this.offset,
+      this.onCreate,
+      this.onUpdate,
+      this.onDelete}) {
+    if (onCreate == null && onUpdate == null && onDelete == null) {
+      throw Exception(
+          "You must provide at least one callback for onCreate, onUpdate or onDelete");
+    }
+  }
+
+  CollectionMetadata _collectionMetadataFromClass(ClassMirror collectionType) {
+    final CollectionMetadata collectionMetadata = collectionType.metadata
+            .firstWhere((element) => element is CollectionMetadata)
+        as CollectionMetadata;
+
+    return collectionMetadata;
+  }
+
+  String toJson() {
+    Map<String, Object> result = {
+      "type": "subscribe",
+      "collection": collectionMetadata.endpointName,
+    };
+    List<String> fieldsAsJson = fieldsToJson;
+    Map<String, dynamic>? filterAsJson = filterToJson;
+
+    final Map<String, dynamic> query = {};
+    query["fields"] = fieldsAsJson;
+
+    if (filterAsJson != null) {
+      query["filter"] = filterAsJson;
+    }
+
+    if (sort != null && sort!.isNotEmpty) {
```

Comment:
no ! please :)

Reply:
I don't see what you are talking about. ;-)

done in bfa2c9cc3a8c5c4f91a2a0ec953631c42602edcf

Reply:
line 68 : `sort!.isNotEmpty`